ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/Assertions.scala

```diff
@@ -34,17 +35,29 @@ object Assertions extends DiffExtensions {
       )
   }
 
-  def assertGrpcError(t: Throwable, expectedCode: Status.Code, pattern: String): Unit =
+  /** Match the given exception against a status code and a regex for the expected message.
+      Succeeds if the exception is a GrpcException with the expected code and
+      the regex matches some part of the message or there is no message and the pattern is
+      empty.
+    */
+  def assertGrpcError(t: Throwable, expectedCode: Status.Code, pattern: Pattern): Unit =
```

Comment:
Just realized you can use `Regex` from Scala, rather than Java's `Pattern`. This allows you to use `.r` to compile a regular expression.

```scala
"my\\s+regex [Ii][Ss] GREAT[!\\.]".r
```

Not too fussed which you prefer.

Reply:
Thanks for the tip, I’ll stick to the Java version for now but happy to change that later.